build_scripts/docker_image_creator.py

```diff
@@ -0,0 +1,65 @@
+"""
+Script to run docker image creation
+Run the command:
+    python docker_image_creator.py docker_file_github_link optimized_binary_link gpu|cpu tensorflow_version python_version
+"""
+import argparse
+import os
+
+def create_docker_image(dockerfile_github_link, optbin_link, processor, framework_version, python_version):
+    """
+    Function builds a docker image with the TF optimized binary
+
+    :param dockerfile_github_link: link to docker file repo
+    :param optbin_link: link to where the optimized binary is
+    :param processor: gpu or cpu
+    :param framework_version: tensorflow version i.e 1.6.0
+    :param python_version: version of python to build container with i.e. 3.6.5 or 2.7.4
+    :return: final built imageid
+    """
+    # 0.) Initialize some commonly used variables
+    dockerfile_repo_name = dockerfile_github_link.split('/')[-1].split('.')[0]
+    pyV = "py{}".format(python_version.split('.')[0]) # i.e. py2
+    framework = "tensorflow"
+    # 1.) Clone dockerfile repo from Github
+    print("Cloning dockerfile repo...")
+    os.system("git clone {}".format(dockerfile_github_link))
+    # 2.) Get optimized binary - need to better parametrize the name
+    print("Getting optimized binary...")
+    optbin_filename = "{}-{}-cp27-cp27mu-manylinux1_x86_64.whl".format(framework, framework_version) # just a tmp file name
+    os.system("curl {} > {}".format(optbin_link, optbin_filename))
+    # 3.) Build base image
+    print("Building base image...")
+    image_name = "{}-base:{}-{}-{}".format(framework, framework_version, processor,  pyV)
+    base_docker_path = "docker/{}/base/Dockerfile.{}".format(framework_version, processor)
+    os.system("sudo nvidia-docker build -t {} -f {}/{} .".format(image_name, dockerfile_repo_name, base_docker_path))
```

Comment:
Is nvidia-docker the only way of building a docker image out of a dockerfile?

Reply:
For our purposes when building for gpus we need to use it.

Reply:
Ahhh, interesting, I was under the impression that gpu images can be built with just regular docker. What has exactly changed for building gpu images?

Reply:
Running some preliminary experiments it looks like docker should work too as well as nvidia-docker. I will look into it further.